### PR TITLE
Swift 3 conversion

### PR DESCRIPTION
**Pull request by lucastrazzullo**

### CODE REVIEW

File: MongoLabKit/Sources/Classes/MongoLabClient.swift [hunk 1]
```diff
@@ -48,20 +48,22 @@ extension MongoLabClient {
...
-        let dataTask = NSURLSession.sharedSession().dataTaskWithRequest(request) {
-            [weak self] (data: NSData?, response: NSURLResponse?, error: NSError?) in
+    fileprivate func dataTask(with request: URLRequest, completion: @escaping Completion) -> URLSessionTask {
+        let dataTask = URLSession.shared.dataTask(with: request) {
+            [weak self] data, response, error in
 
-            self?.parseData(data, response: response, error: error, completion: completion)
+            self?.parse(data, response: response, error: error, completion: completion)
+
+            return
```
Comment (madhikarma) [inline MongoLabKit/Sources/Classes/MongoLabClient.swift]: Hey, just curious but why is the return necessary?

Also just an idea but you can bind self as follows 

`if let strongSelf = self`

but optional works too of course

File: MongoLabKit/Sources/Classes/MongoLabError.swift [hunk 2]
```diff
@@ -9,22 +9,22 @@
...
 
 
     public func description() -> String {
         switch self {
-        case .ConnectionError:
+        case .connectionError:
             return "Unable to connect to server."
-        case .RequestError:
+        case .requestError:
             return "Unable to create a request to the database. baseURL and apiKey MUST be not empty in the configuration."
```
Comment (madhikarma) [inline MongoLabKit/Sources/Classes/MongoLabError.swift]: Not sure if these are user facing but perhaps we can bundle these into a Localisable.strings bundle to be embedded into the framework. Not related to the Swift 3 migration though

**Comment by madhikarma** [comment]
Code review complete @lucastrazzullo just a question from me but looks good. Nice work

File: MongoLabKit/Sources/Classes/MongoLabError.swift (hunk 2 above)
Comment (lucastrazzullo) [inline MongoLabKit/Sources/Classes/MongoLabError.swift]: Yeah definitely, just put on the backlog :)

File: MongoLabKit/Sources/Classes/MongoLabClient.swift (hunk 1 above)
Comment (lucastrazzullo) [inline MongoLabKit/Sources/Classes/MongoLabClient.swift]: Well in this case it's just a taste decision, tried to keep it cleaner without adding a condition to unwrap the value, I reckon all of those things will change when I'll refactor the syntax to conform to the open source code conventions

**Pull request merged**